CMakeLists.txt

```diff
@@ -36,11 +38,14 @@ target_compile_definitions(
 set_target_properties(
     tinyxml2
     PROPERTIES
-    DEFINE_SYMBOL "TINYXML2_EXPORT"
     VERSION "${tinyxml2_VERSION}"
     SOVERSION "${tinyxml2_VERSION_MAJOR}"
 )
 
+if(BUILD_SHARED_LIBS)
+    target_compile_definitions(tinyxml2 PRIVATE "TINYXML2_EXPORT")
+endif()
+

```

Comment:
@aminya - this whole change is pointless. Please see my other comment.

Reply:
I want to make a shared library file that uses tinyxml2 internally but doesn't export the symbols of tinyxml2. How is that possible without my proposal?

Reply:
I think you can accomplish that by only changing `tinyxml2.h` in the way you did. The change I selected here is a no-op. I do not believe you need to touch `CMakeLists.txt`.

Reply:
[`093dd7a` (#922)](https://github.com/leethomason/tinyxml2/pull/922/commits/093dd7a1fe484c58c63a0e7b5a9a5f08b4efef5e)